src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import { useMemo } from 'react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { useAuth } from '@/utils/AuthProvider';
+import { useProfile } from '@/utils/ProfileProvider';
+import IJPlogo from '~/public/images/ijp-logo.webp';
+import { ProfileButton, SmallLinkButton } from '../Buttons';
+import * as Styles from './style';
+
+export default function NavBar() {
+  const profile = useProfile();
+  if (!profile) throw new Error('Profile must be defined.');
+
+  const auth = useAuth();
+  if (!auth) throw new Error('Auth Must be defined.');
+
+  const AuthButtonView = useMemo(() => {
```

Comment:
let's use camel case for this since this is not technically a component constructor (but rather just TSX)